modules/core/src/bitgo.ts

```diff
@@ -141,171 +149,437 @@ const createResponseErrorString = function(res) {
+}
+
+interface SplitSecretOptions {
+  seed: string;
+  passwords: string[];
+  m: number;
+}
+
+interface SplitSecret {
+  xpub: string;
+  m: number;
+  n: number;
+  seedShares: any;
+}
+
+interface ReconstituteSecretOptions {
+  shards: string[];
+  passwords: string[];
+}
+
+interface ReconstitutedSecret {
+  xpub: string;
+  xprv: string;
+  seed: string;
+}
+
+interface VerifyShardsOptions {
+  shards: string[];
+  passwords: string[];
+  m: number;
+  xpub: string;
+}
+
+interface GetEcdhSecretOptions {
+  otherPubKeyHex: string;
+  eckey: { d: any };
+}
+
+interface AccessTokenOptions {
+  accessToken: string;
+}
+
+interface TokenIssuanceResponse {
+  derivationPath: string;
+  encryptedToken: string;
+  encryptedECDHXprv?: string;
+}
+
+interface TokenIssuance {
+  token: string;
+  ecdhXprv?: string;
+}
+
+interface CalculateHmacSubjectOptions {
+  urlPath: string;
+  text: string;
+  timestamp: number;
+  statusCode?: number;
+}
+
+interface CalculateRequestHmacOptions {
+  url: string;
+  text: string;
+  timestamp: number;
+  token: string;
+}
+
+interface CalculateRequestHeadersOptions {
+  url: string;
+  text: string;
+  token: string;
+}
+
+interface RequestHeaders {
+  hmac: string;
+  timestamp: number;
+  tokenHash: string;
+}
+
+interface VerifyResponseOptions extends CalculateRequestHeadersOptions {
+  hmac: string;
+  url: string;
+  text: string;
+  timestamp: number;
+  statusCode?: number;
+}
+
+interface AuthenticateOptions {
+  username: string;
+  password: string;
+  otp?: string;
+  trust?: number;
+  forceSMS?: boolean;
+  extensible?: boolean;
+  forceV1Auth?: boolean;
+}
+
+interface ProcessedAuthenticationOptions {
+  email: string;
+  password: string;
+  forceSMS: boolean;
+  otp?: string;
+  trust?: number;
+  extensible?: boolean;
+  extensionAddress?: string;
+  forceV1Auth?: boolean;
+}
+
+interface AddAccessTokenOptions {
+  label: string;
+  otp?: string;
+  duration?: number;
+  ipRestrict?: string[];
+  txValueLimit?: number;
+  scope: string[];
+}
+
+interface RemoveAccessTokenOptions {
+  id?: string;
+  label?: string;
+}
+
+interface GetUserOptions {
+  id: string;
+}
+
+interface ChangePasswordOptions {
+  oldPassword: string;
+  newPassword: string;
+}
+
+interface UnlockOptions {
+  otp?: string;
+  duration?: number
+}
+
+interface ExtendTokenOptions {
+  duration?: string;
+}
+
+interface GetSharingKeyOptions {
+  email: string;
+}
+
+interface PingOptions {
+  reqId?: RequestTracer;
+}
+
+interface EstimateFeeOptions {
+  numBlocks?: number;
+  maxFee?: number;
+  inputs?: string[];
+  txSize?: number;
+  cpfpAware?: boolean;
+}
+
+interface WebhookOptions {
+  url: string;
+  type: string;
+}
+
+interface ListWebhookiNotificationsOptions {
+  prevId?: string;
+  limit?: number;
+}
+
+interface SimulateWebhookOptions {
+  webhookId: string;
+  blockId: string;
+}
+
+export interface BitGo {
+  get(url: string, callback?: NodeCallback<superagent.Response>): superagent.Request;
```

Comment:
shouldn't the return type for```get``` be a ```Response``` rather than a ```Request```?

Reply:
No, you need to resolve the request into a response by yielding/awaiting it (it's like a promise)

Reply:
ah ok